voila/app.py

```diff
@@ -7,78 +7,56 @@
 
-from zmq.eventloop import ioloop
-
 import gettext
-import io
-import json
 import logging
-import threading
-import tempfile
 import os
-import shutil
-import signal
-import socket
-import webbrowser
-import errno
-import random
-
-try:
-    from urllib.parse import urljoin
-    from urllib.request import pathname2url
-except ImportError:
-    from urllib import pathname2url
-    from urlparse import urljoin
 
 import jinja2
 
-import tornado.ioloop
-import tornado.web
-
-from traitlets.config.application import Application
-from traitlets.config.loader import Config
-from traitlets import Unicode, Integer, Bool, Dict, List, default
+from traitlets import Unicode, Integer, Bool, Dict, List
 
-from jupyter_server.services.kernels.kernelmanager import AsyncMappingKernelManager
-from jupyter_server.services.kernels.handlers import KernelHandler, ZMQChannelsHandler
-from jupyter_server.services.contents.largefilemanager import LargeFileManager
+from jupyter_server.extension.application import ExtensionApp
+from jupyter_server.serverapp import ServerApp
 from jupyter_server.base.handlers import FileFindHandler, path_regex
-from jupyter_server.config_manager import recursive_update
-from jupyter_server.utils import url_path_join, run_sync
-from jupyter_server.services.config import ConfigManager
-
-from jupyter_client.kernelspec import KernelSpecManager
-
 from jupyter_core.paths import jupyter_config_path, jupyter_path
 
-from ipython_genutils.py3compat import getcwd
-
 from .paths import ROOT, STATIC_ROOT, collect_template_paths
 from .handler import VoilaHandler
 from .treehandler import VoilaTreeHandler
+from .static_file_handler import WhiteListFileHandler
 from ._version import __version__
-from .static_file_handler import MultiStaticFileHandler, WhiteListFileHandler
-from .configuration import VoilaConfiguration
 from .execute import VoilaExecutor
 from .exporter import VoilaExporter
 from .csspreprocessor import VoilaCSSPreprocessor
 
-ioloop.install()
 _kernel_id_regex = r"(?P<kernel_id>\w+-\w+-\w+-\w+-\w+)"
 
 
 def _(x):
     return x
 
 
-class Voila(Application):
-    name = 'voila'
+class Voila(ExtensionApp):
+    extension_name = 'voila'
+    extension_url = '/voila'
+    load_other_extensions = True
+
     version = __version__
-    examples = 'voila example.ipynb --port 8888'
+    examples = 'jupyter voila example.ipynb'
 
     flags = {
-        'debug': ({'Voila': {'log_level': logging.DEBUG}}, _("Set the log level to logging.DEBUG")),
-        'no-browser': ({'Voila': {'open_browser': False}}, _('Don\'t open the notebook in a browser after startup.'))
+        'debug': (
+            {'ServerApp': {'log_level': logging.DEBUG}},
+            _("Set the log level to logging.DEBUG")
+        ),
+        'no-browser': (
+            {'Voila': {'open_browser': False}},
+            _("Don't open the notebook in a browser after startup.")
+        ),
+        "standalone": (
```

Comment:
It seems this disappeared, but I am not sure how we can detect if we are in 'standalone' mode, do you know?

Reply:
It think this is replaced by `load_other_extensions`, which I set to `True` because of this issue: https://github.com/jupyter/jupyter_server/issues/211